Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+
+The Roadmap Document contains the following information:
+- List of current milestones
+  - Sequentally ordered, eg. by expected due date (where applicable)
+  - Next upcoming milestone at the top
+- Current status of the project, ie. milestone progress
+- List of previous milestones
+
+The roadmap can be a high-level overview of milestones and project's progress, so details such as list of goals per milestone can be omitted. However, if omitted, the roadmap must contain links to the detailed break down.
+
+Examples:
+- [js-ipfs roadmap](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#ipfs-javascript-implementation-roadmap)
+- [Orbit roadmap](https://github.com/haadcode/orbit/blob/master/ROADMAP.md#orbit---roadmap)
+
+### Organization Roadmap
+An overview of all projects across the organization is tracked in organization's roadmap. This is similar to a project roadmap, but collects all milestones from all projects into one place.
+
+Organization roadmap contains the following information:
+- List of all projects
+- List of all current milestones and their progress in each project
+- List of links to old milestones
+
+The organization roadmap is owned by project management and project managers are responsible for generating and keeping the roadmap up to date. The roadmap should be updated in the quarterly planning meeting as old milestones are reviewed and the project leads commit to new milestones.
+
+TODO: a clear example
+
+## Workflow
+
+### Pipelines
+The IPFS Project Management Process uses Pipelines to process goals towards a release. Pipelines are a simple way to describe the workflow of a project and move Goals between the different stages in their lifecycle. 
+
+#### Stages
+Pipelines are sequential stages through which a Goal goes during its lifetime. The different stages are broken down per the workflow of each project. On high level the stages can be described as: `Not Done ──> In Progress ──> Done`. Different projects might have different stages in their development pipeline and it is up to the project lead to identify and define the stages her project uses.
+
+***Proposal: Unify "Future Work", "Backlog" and "Done" stages for all projects, let project leads define stages of the "Development" phase.***
+
+#### Flow
+Goals move from pipeline's entry stage towards the completion stage and a Goal can be only in one stage at a time. Usually goals move one way, in sequential order, but sometimes goals can move back between the stages (eg. `In Progress ──> Review ──> In Progress`). Sometimes a goal can't progress due to something preventing it to move to the next stage. To ensure surfacing blockers, each pipeline should have a ***"Blocked"*** stage where blocked goals are moved as soon as they occur. 
```

Comment:
I think we need to drill into the terminology here again as I feel there's a difference of perspective to its meaning. 

I think you're coming more from the perspective of "everything is a pipeline", eg. non-development pipelines such as "hiring pipeline", meaning the full process of a particular area. Am I correct?

What I mean with a pipeline in this context is the mechanics of how *value* gets produced within a project, ie. how a goal gets from the backlog to a release, to "not done" to "done". It's the "assembly line" or the "production line" of *producing* an artifact.

> Some pipelines are not for tracking blockages, or progress at all.

Can you give an example what you're thinking here? I see it the opposite: every pipeline, while not (explicitly) **for** tracking progress, gives you progress information.

> No, this does not fit ALL pipelines ever.

That's why it's *should* and not *must*. If you consider a pipeline a away to process goals and tasks, then it's highly beneficial to make sure the pipeline processes those goals as fast as possible, no? A blockage creates friction and causes the pipeline to output slower, so it is in your best interest to know asap and as clearly as possible if something is *not* moving, right?

> En example of a pipeline NOT to track "development progress"

As an example here, while you could consider the "Block bad bits from Gateway" as a pipeline, it could also be considered a checklist, eg. a quality criteria to move to the next stage. As a production pipeline, you could simplify the stages to: Backlog ("Receive report") -> "In Progress" (from "validate" to "Test Block" ) -> "Done".